### PR TITLE
Feature/gsye 636

### PR DESCRIPTION
**Pull request by spyrostz**

### CODE REVIEW

File: gsy_framework/utils.py [hunk 1]
```diff
@@ -471,6 +471,13 @@ def convert_kW_to_kWh(power_W, slot_length):
     return convert_W_to_Wh(power_W, slot_length)
 
 
+# pylint: disable=invalid-name
+def convert_kWh_to_W(energy_kWh, slot_length):
+    """Convert W power into energy in Wh (based on the duration of the market slot)."""
+    energy_Wh = energy_kWh * 1000.0
+    return energy_Wh / (slot_length / duration(hours=1))
```
Comment (hannesdiedrich) [inline gsy_framework/utils.py:478]: which unit or type has the `slot_length` ? From the code, I see that it should be `pendulum.duration`, right?
If so, I propose 
```suggestion
def convert_kWh_to_W(energy_kWh: float, slot_length: duration) -> float:
    """Convert W power into energy in Wh (based on the duration of the market slot)."""
    energy_Wh = energy_kWh * 1000.0
    return energy_Wh / slot_length.total_hours()
```

File: gsy_framework/constants_limits.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ class HeatPumpSettings:
         """Default values for the heat pump."""
         # range limits
         MAX_POWER_RATING_KW_LIMIT = RangeLimit(0, sys.maxsize)
-        TEMP_C_LIMIT = RangeLimit(-273.15, 6000)
+        TEMP_C_LIMIT = RangeLimit(0.0, 200.0)
```
Comment (hannesdiedrich) [inline gsy_framework/constants_limits.py:189]: :-D 
Finally somebody reacted to my easter egg.

File: gsy_framework/validators/heat_pump_validator.py [hunk 3]
```diff
@@ -46,17 +57,12 @@ def _validate_temp(cls, **kwargs):
 
         min_temp_c = kwargs["min_temp_C"]
         max_temp_c = kwargs["max_temp_C"]
+        initial_temp_c = kwargs["initial_temp_C"]
 
-        if not min_temp_c <= max_temp_c:
+        if not min_temp_c <= initial_temp_c <= max_temp_c:
```
Comment (hannesdiedrich) [inline gsy_framework/validators/heat_pump_validator.py:62]: great! Thank you!

File: gsy_framework/validators/heat_pump_validator.py [hunk 4]
```diff
@@ -78,8 +84,8 @@ def validate_rate(cls, **kwargs):
         for buying_rate_arg_name in buying_rate_arg_names:
             cls._check_range(
                 name=buying_rate_arg_name, value=kwargs[buying_rate_arg_name],
-                min_value=HeatPumpSettings.TEMP_C_LIMIT.min,
-                max_value=HeatPumpSettings.TEMP_C_LIMIT.max)
+                min_value=HeatPumpSettings.BUYING_RATE_LIMIT.initial,
```
Comment (hannesdiedrich) [inline gsy_framework/validators/heat_pump_validator.py:87]: Upsi

**Review by hannesdiedrich — DISMISSED** [review]
I have only jabbering comments. LGTM

File: gsy_framework/utils.py (hunk 1 above)
Comment (spyrostz) [reply gsy_framework/utils.py:478]: I added the type hints, however did not follow the `total_hours` suggestion, in order to be more evident on how the kWh -> W conversion is done here. Thanks!

Comment (hannesdiedrich) [reply gsy_framework/utils.py]: I just wanted to simplify the `(slot_length / duration(hours=1))`.
IMO we could use the features pendulum comes with. 
But if you do not like this, I am convinced to leave it.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**